app/contexts/amplience/index.js

```diff
@@ -90,29 +105,56 @@ const getCookies = (headers) => {
     return {}
 }
 
-export const generateVseProps = ({req, res, query}) => {
-    let vse = null
-    let vseTimestamp = 0
+export const getGroupsFromLocalStorage = ({req, res}) => {
+    let groups = []
 
     if (res) {
-        vse = query.vse
-        vseTimestamp = query['vse-timestamp']
-        vseTimestamp = vseTimestamp == 'null' ? 0 : Number(vseTimestamp)
-
-        if (vse == null) {
-            const cookies = getCookies(req.rawHeaders)
-            vse = cookies['vse']
-            vseTimestamp = cookies['vse-timestamp']
-            vseTimestamp = vseTimestamp != null ? Number(vseTimestamp) : undefined
+        const cookies = getCookies(req.rawHeaders)
+        groups = cookies['customerGroups']
+
+        try {
+            groups = JSON.parse(groups)
+        } catch (e) {
+            console.error(`No customer groups found in cookies`)
         }
     }
 
-    return {vse, vseTimestamp}
+    return groups
+}
+
+export const generateVseProps = ({req, res, query}) => {
+    // '/:locale/visualization/:hubname/:contentId/:vse'
+    const vizRegEx = /\/(.*)\/visualization\/(.*)\/(.*)\/(.*)/
+    if (req.originalUrl.match(vizRegEx)) {
+        const url = req.originalUrl.split('?')[0]
+        const [match, locale, hubname, contentId, vse] = url.match(vizRegEx)
+        return {vse, hubname, contentId, locale}
+    } else if (query['vse'] || query['vse-timestamp']) {
+        let vse = null
+        let vseTimestamp = 0
+
+        // why if (res)? we don't use res anywhere here?
```

Comment:
I think `res` is not present if this method is called from the clientside. _app index shouldn't be called from there anyways.

Reply:
Removed comment